simple/testdata/src/LintRedundantCanonicalHeaderKey/LintRedundantCanonicalHeaderKey.go

```diff
@@ -0,0 +1,16 @@
+package pkg
+
+import "net/http"
+
+func fn1() {
```

Comment:
Just to be thorough, add test cases for the following non-matches:

- using http.CanonicalHeaderKey on the value instead of the key
- using something like `fn(http.CanonicalHeaderKey(...))` as the key
- using something like `http.CanonicalHeaderKey(...) + ...` as the key